app/assets/javascripts/discourse/app/components/sidebar/messages-section.js

```diff
@@ -37,27 +44,38 @@ export default class SidebarMessagesSection extends GlimmerComponent {
     currentRouteParentName,
     currentRouteParams,
   }) {
-    const sectionLinks = [
-      ...this.personalMessagesSectionLinks,
-      ...this.groupMessagesSectionLinks,
-    ];
-
-    if (currentRouteParentName !== "userPrivateMessages") {
-      sectionLinks.forEach((sectionLink) => {
+    if (
+      currentRouteParentName !== "userPrivateMessages" &&
+      currentRouteParentName !== "topic"
+    ) {
+      for (const sectionLink of this.allSectionLinks) {
         sectionLink.collapse();
-      });
+      }
     } else {
-      sectionLinks.forEach((sectionLink) => {
-        sectionLink.pageChanged(currentRouteName, currentRouteParams);
-      });
+      const attrs = {
+        currentRouteName,
+        currentRouteParams,
+      };
+
+      if (currentRouteParentName === "topic") {
+        const topicController = getOwner(this).lookup("controller:topic");
+
+        if (topicController.model.isPrivateMessage) {
```

Comment:
Is topicController and .model always guaranteed to exist? May need safe nav if not

Reply:
Yup because we've check that the parent route is a `topic` route so the controller and model will always exist.